src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -74,6 +74,18 @@ public interface IScheduler
         IEnumerable<AsyncTask> Tasks { get; }
     }
 
+    public enum TaskProcessMode
+    {
+        /// <summary>
+        /// Scheduled task will be processed immediately
+        /// </summary>
+        Immediately,
+        /// <summary>
+        /// Scheduled task will be processed by schedule
+        /// </summary>
+        BySchedule
+    }
+
```

Comment:
`Immediately` seems to be more of an adverb and `BySchedule` is a noun form. Wouldn't it be easier to use a more conventional/familiar term like `Synchronous` and `Asynchronous`?